### PR TITLE
Replace Array class with numpy-based Vector class

### LINKED ISSUE #117: kwave.Array operator support ( + - * / etc...)
Something like this could be helpful:

```python
import numpy as np

class Vector(np.ndarray):
    def __new__(cls, x, y, z):
        obj = np.array([x, y, z], dtype=float).view(cls)
        obj.x = obj[0]
        obj.y = obj[1]
        obj.z = obj[2]
        return obj
    
    def __array_finalize__(self, obj):
        if obj is None: return
        self.x = getattr(obj, 'x', None)
        self.y = getattr(obj, 'y', None)
        self.z = getattr(obj, 'z', None)

```

### PR DESCRIPTION
**Pull request by faridyagubbayli**
Closes #117 

This PR introduces the new `Vector` class, an enhanced version of the previous `Array` class. As a subclass of `numpy.ndarray`, the `Vector` class enables mathematical operations on its data. However, some use cases in the repository cannot be represented by a `numpy` array due to its inherent constraints, such as the inability to support elements with varying sizes within an array. To address this limitation, the existing `Array` class has been retained but renamed to `FlexibleVector`.

### CODE REVIEW

File: kwave/__init__.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 # is refactored
 
 platform = sys.platform
+platform = 'linux'
```
Comment (waltsims) [inline kwave/__init__.py:18]: This was for local debugging no? and can be removed from the PR?

File: kwave/data.py [hunk 2]
```diff
@@ -1,10 +1,55 @@
 from dataclasses import dataclass
+from typing import Any
 
 import numpy as np
 
 
+class Vector(np.ndarray):
+    def __new__(cls, elements: list):
+        elements = list(elements)[:3]  # Truncate the input list to have at most 3 elements
```
Comment (waltsims) [inline kwave/data.py:9]: would it be better to assert input list has 3 elements?

File: kwave/kgrid.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 import numpy as np
 
-from kwave.data import Array
+from kwave.data import Vector, FlexibleVector
```
Comment (waltsims) [inline kwave/kgrid.py:7]: I don't understand the difference in use-case between Vector and FlexibleVector at first glance.

**Review by waltsims — CHANGES_REQUESTED** [review]
This is an exciting PR

File: kwave/data.py [hunk 4]
```diff
@@ -1,10 +1,55 @@
...
+        new_coordinates = list(self) + [value]
+        return Vector(new_coordinates)
+
+
 @dataclass
-class Array(object):
+class FlexibleVector(object):
     """
         This class is very similar to Numpy.ndarray but there are differences:
             - It can have 3 elements at max
```
Comment (waltsims) [inline kwave/data.py:55]: Add comment that this class will be deprecated after kGrid is refactored.

**Review by waltsims — CHANGES_REQUESTED** [review]

File: tests/matlab_test_data_collectors/python_testers/utils/check_equality.py [hunk 5]
```diff
@@ -49,6 +49,7 @@ def check_kgrid_equality(kgrid_object: kWaveGrid, expected_kgrid_dict: dict):
         else:
             ignore_if_nan = False
 
+        print(mapped_key)
```
[the remark this answers is not in the document]
Comment (faridyagubbayli) [reply tests/matlab_test_data_collectors/python_testers/utils/check_equality.py]: Yess, removed it

File: kwave/data.py (hunk 2 above)
Comment (faridyagubbayli) [reply kwave/data.py]: Added the assertion and removed logic to take the first three elements

File: kwave/__init__.py (hunk 1 above)
Comment (faridyagubbayli) [reply kwave/__init__.py]: Good catch! Removed it

File: kwave/kgrid.py (hunk 3 above)
Comment (faridyagubbayli) [reply kwave/kgrid.py:7]: We had a offline discussion after the comment but putting the idea here for posterity. `Vector` is designed to contain elements of the same type while `FlexibleVector` can contain mixture of element types. Ideally, we don't want `FlexibleVector` functionality. It is deprecated and will be removed once we refactor the `kWaveGrid` object. It is the only place where `FlexibleVector` is used.

File: kwave/data.py (hunk 4 above)
Comment (faridyagubbayli) [reply kwave/data.py:56]: :+1: Done!

**Review by waltsims — APPROVED** [review]

**Pull request merged**